src/lib.rs

```diff
@@ -9,6 +9,94 @@
 // disable warnings in that scenario.
 #![cfg_attr(not(rayon_unstable), allow(warnings))]
 
+//! Data-parallelism library that is easy to convert sequential computations into parallel.
+//!
+//! Rayon is lightweight and convenient for application to existing code.  It guarantees
+//! data-race free executions, and takes advantage of parallelism when sensible, based
+//! on work-load at runtime.  There are two categories of rayon workloads: parallel
+//! iterators and multi-branched recursion (`join` method).
+//!
+//! # Parallel Iterators
+//!
+//! Parallel iterators are formed using [`par_iter`] and `par_iter_mut` functions
```

Comment:
It's strange to me that you've linked `par_iter` to the `ParallelIterator` trait.  It is good to have that link up front, but `par_iter` itself comes from `IntoParallelRefIterator`.  There are actually three such conversions, `into_par_iter()`, `par_iter()`, and `par_iter_mut()`, to iterate respectively by value, by shared reference, or by mutable reference.